packages/wonder-blocks-modal/components/modal-launcher.js

```diff
@@ -137,17 +146,24 @@ export default class ModalLauncher extends React.Component<Props, State> {
                         /* We need the container View that FocusTrap creates to be at the
                            correct z-index so that it'll be above the global nav in webapp. */
                         <FocusTrap style={styles.container}>
-                            <ModalBackdrop onCloseModal={this.handleCloseModal}>
+                            <ModalBackdrop
+                                onCloseModal={
+                                    this.props.backdropDismissEnabled
+                                        ? this.handleCloseModal
+                                        : () => {}
+                                }
+                            >
                                 {this._renderModal()}
                             </ModalBackdrop>
                         </FocusTrap>,
                         body,
                     )}
-                {this.state.opened && (
-                    <ModalLauncherKeypressListener
-                        onClose={this.handleCloseModal}
-                    />
-                )}
+                {this.state.opened &&
+                    this.props.backdropDismissEnabled && (
+                        <ModalLauncherKeypressListener
+                            onClose={this.handleCloseModal}
```

Comment:
It could work nicely for the first case (line #151), but for this case, I think this could lead to unnecessarily instantiate the `<ModalLauncherKeypressListener />` renderless component. Not a big deal, tbh as we could pass `backdropDismissEnabled` prop down to this component and add some extra validations to avoid adding the event handler on line # 181.

What do you think?